zjit/src/codegen.rs

```diff
@@ -101,6 +101,9 @@ pub extern "C" fn rb_zjit_iseq_gen_entry_point(iseq: IseqPtr, _ec: EcPtr) -> *co
 /// See [gen_iseq_entry_point_body]. This wrapper is to make sure cb.mark_all_executable()
 /// is called even if gen_iseq_entry_point_body() partially fails and returns a null pointer.
 fn gen_iseq_entry_point(iseq: IseqPtr) -> *const u8 {
+    if !ZJITState::can_compile_iseq(iseq_get_location(iseq, 0)) {
```

Comment:
Can you avoid allocating a String here when `--zjit-allowed-iseqs` is not in use? 

In case you missed it, `asm_comment!(asm, "ZJIT entry point: {}", iseq_get_location(iseq, 0));` doesn't allocate any string when `--zjit-dump-disasm` is not enabled.